src/ClinicalDecisionRules/Interface/RuleLibrary/RuleManager.php

```diff
@@ -494,62 +508,46 @@ function deleteRuleFilter($rule, $guid)
         sqlStatement("DELETE FROM rule_filter WHERE SHA1(CONCAT( id, include_flag, required_flag, method, method_detail, value )) = ?", [$guid]);
     }
 
-    function updateSummary($ruleId, $types, $title, $developer, $funding, $release, $web_ref, $bibliographic_citation, $linked_referential_cds)
+    public function updateSummaryForRule(Rule $rule)
     {
-        $rule = $this->getRule($ruleId);
 
-        // if $types is empty, then set to empty array so the in_array() calls below will work
-        $types = $types ?: [];
-
-        if (is_null($rule)) {
-            // add
-            $result = sqlQuery("select count(*)+1 AS id from clinical_rules");
-            $ruleId = "rule_" . $result['id'];
-            sqlStatement(
-                "INSERT INTO clinical_rules (id, pid, active_alert_flag, passive_alert_flag, cqm_flag, amc_flag, patient_reminder_flag, developer, funding_source, release_version, web_reference, bibliographic_citation, linked_referential_cds ) " .
-                    "VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?) ",
-                array(
-                        $ruleId,
-                        0,
-                        in_array(RuleType::ActiveAlert, $types) ? 1 : 0,
-                        in_array(RuleType::PassiveAlert, $types) ? 1 : 0,
-                        in_array(RuleType::CQM, $types) ? 1 : 0,
-                        in_array(RuleType::AMC, $types) ? 1 : 0,
-                        in_array(RuleType::PatientReminder, $types) ? 1 : 0,
-                        $developer,
-                        $funding,
-                        $release,
-                        $web_ref,
-                        $bibliographic_citation,
-                        $linked_referential_cds
-                    )
-            );
-
-            // do label
-            $this->doRuleLabel(false, "clinical_rules", $ruleId, $title);
-            return $ruleId;
+        $exists = !empty($rule->id);
+        if (!$exists) {
+            $dataToPersist = $rule->getSummaryDataToPersist();
+            // seems odd to have this here.
+            $dataToPersist['id'] = $this->getNextRuleId();
+            $dataToPersist['pid'] = 0;
+            $values = str_repeat("?,", count($dataToPersist) - 1) . "?";
+            $sql = "INSERT INTO `clinical_rules` (" . implode(",", array_keys($dataToPersist))
```

Comment:
just making sure escaping not needed on $dataToPersist keys

Reply:
Nope, column names are hard-coded and do not come from user data so we are safe here.